pcsx2/x86/microVU_Macro.inl

```diff
@@ -55,24 +55,38 @@ void setupMacroOp(int mode, const char* opName)
 	{
 		xMOVSSZX(xmmPQ, ptr32[&vu0Regs.VI[REG_Q].UL]);
 	}
-	if (mode & 0x08) // Clip Instruction
+	if (mode & 0x08 && (!CHECK_COP2_FLAGHACK || g_pCurInstInfo->info & EEINST_COP2_CLIP_FLAG)) // Clip Instruction
 	{
 		microVU0.prog.IRinfo.info[0].cFlag.write     = 0xff;
 		microVU0.prog.IRinfo.info[0].cFlag.lastWrite = 0xff;
 	}
-	if (mode & 0x10) // Update Status/Mac Flags
+	if (mode & 0x10 && (!CHECK_COP2_FLAGHACK || g_pCurInstInfo->info & EEINST_COP2_STATUS_FLAG)) // Update Status Flag
 	{
 		microVU0.prog.IRinfo.info[0].sFlag.doFlag      = true;
 		microVU0.prog.IRinfo.info[0].sFlag.doNonSticky = true;
 		microVU0.prog.IRinfo.info[0].sFlag.write       = 0;
 		microVU0.prog.IRinfo.info[0].sFlag.lastWrite   = 0;
+	}
+	if (mode & 0x10 && (!CHECK_COP2_FLAGHACK || g_pCurInstInfo->info & EEINST_COP2_MAC_FLAG)) // Update Mac Flags
+	{
 		microVU0.prog.IRinfo.info[0].mFlag.doFlag      = true;
 		microVU0.prog.IRinfo.info[0].mFlag.write       = 0xff;
-		_freeX86reg(ebx);
-		//Denormalize
-		mVUallocSFLAGd(&vu0Regs.VI[REG_STATUS_FLAG].UL);
-		
-		xMOV(gprF0, eax);
+	}
+	if (mode & 0x10)
+	{
+		if (!CHECK_COP2_FLAGHACK || (g_pCurInstInfo->info & EEINST_COP2_DENORMALIZE_STATUS_FLAG))
+		{
+			_freeX86reg(ebx);
```

Comment:
should this not be outside the if? since ebx == gprF0 and the other case is also modifying it